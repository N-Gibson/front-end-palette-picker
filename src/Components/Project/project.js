import React from 'react';
import { deleteProject } from '../../apiCalls';
import { Link } from 'react-router-dom';
import './project.scss';

const Project = (props) => {
  const { changeName, handleProject } = props;
  return (
    <section className='project' id={`project_${props.id}`}>
      <textarea name='revisedName' onChange={changeName}>{props.name}</textarea>
      <Link to={`/project/${props.id}` }>
        <button onClick={() => handleProject(props.name)}>Show</button>
      </Link>
      <button 
        type='button' 
        onClick={() => {
          deleteProject(props.id);
        }
      }>Delete</button>
    </section>
  )
}

export default Project;
