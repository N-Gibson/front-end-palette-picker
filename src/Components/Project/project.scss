.project {
  display: flex;
  flex-direction: column;
}

.project__button{
  width: 70%;
  height: 3vh;
}

a {
  width: 100%
}