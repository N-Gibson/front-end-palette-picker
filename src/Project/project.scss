main {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 10% 1fr 2fr;
  grid-template-areas:
    'nav'
    'inputs'
    'projects';
  min-height: 100vh;
}

h1 {
  display: flex;
  flex-direction: flex-start;
  grid-area: nav;
  margin-left: 20px;
}

.project-inputs {
  display: flex;
  flex-direction: column;
  grid-area: inputs;
  justify-content: center;
  margin: 0px auto;
  width: 50%;
}

.project-input {
  font-size: 1em;
  height: 50px;
  margin-bottom: 5px;
  padding-left: 10px;
}

.project-button {
  cursor: pointer;
  font-size: 1.2em;
  height: 50px;
  margin-top: 5px;
    &:hover {
      background-color: lightgrey;
    }
}

.project-display {
  align-items: center;
  background-color: #D9DCDF;
  display: grid;
  grid-area: projects;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-gap: 20px;
  padding: 20px;
    &:hover .project-prompt {
      color: black;
    }
}

.project-prompt {
  color: #D9DCDF;
  left: 42%;
  position: absolute;
}

.project {
  align-items: center;
  border: 1px solid;
  cursor: pointer;
  display: flex;
  height: 75px;
  justify-content: center;
   &:hover {
     background-color: #fff;
   }
}
