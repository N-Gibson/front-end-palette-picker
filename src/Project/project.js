import React from 'react';
import './project.scss';

const Project = (props) => {
  console.log(props)
  return (
    <section className='project' id={`project_${props.id}`}>
      <h2>{props.name}</h2>
    </section>
  )
}

export default Project;