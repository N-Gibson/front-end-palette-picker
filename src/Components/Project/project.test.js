import React from 'react';
import { shallow } from 'enzyme';
import Project from './project'

describe('Project', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Project />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})