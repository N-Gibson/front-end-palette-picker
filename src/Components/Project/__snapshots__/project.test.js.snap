// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot 1`] = `
<section
  className="project"
  id="project_undefined"
>
  <h3
    name="revisedName"
  />
  <Link
    to="/project/undefined"
  >
    <button
      className="project__button"
      onClick={[Function]}
    >
      Show
    </button>
  </Link>
  <button
    className="project__button"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
</section>
`;
